mlir/include/mlir/Dialect/ControlFlow/IR/ControlFlowOps.td

```diff
@@ -37,7 +37,7 @@ class CF_Op<string mnemonic, list<Trait> traits = []> :
 // AssertOp
 //===----------------------------------------------------------------------===//
 
-def AssertOp : CF_Op<"assert"> {
+def AssertOp : CF_Op<"assert", [Pure]> {
```

Comment:
I just noticed a problem with this approach. If the op is "pure", it will just fold away. That's why some of the tests are failing. We have to give it some side effect. I'd suggest a "write" effect not attached to any SSA value, with a comment that it models the side effect of "program termination". This can't be done with a trait, you'd have to implement the `MemoryEffectsOpInterface::getEffects`.

Reply:
Good catch. I did not end up running all the tests last night, only my own. I made the changes you suggested. PTAL.

Reply:
@matthias-springer Just following up.